enhancements/cluster-version-operator/upgrades.md

```diff
@@ -0,0 +1,83 @@
+# Upgrades and order
+
+In the CVO, upgrades are performed in the order described in the payload (lexographic), with operators only running in parallel if they share the same file prefix `0000_NN_` and differ by the next chunk `OPERATOR`, i.e. `0000_70_authentication-operator_*` and `0000_70_image-registry-operator_*` are run in parallel.
+
+Priorities during upgrade:
+
+1. Simplify the problems that can occur
+2. Ensure applications aren't disrupted during upgrade
+3. Complete within a reasonable time period (30m-1h for control plane)
+
+During upgrade we bias towards predictable ordering for operators that lack sophistication about detecting their prerequisites. Over time, operators should be better at detecting their prerequisites without overcomplicating or risking the predictability of upgrades.
+
+Currently, upgrades proceed in operator order without distinguishing between node and control plane components. Future improvements may allow nodes to upgrade independently and at different schedules to reduce production impact. This in turn complicates the investment operator teams must make in testing and understanding how to version their control plane components independently of node infrastructure.
```

Comment:
Worth including [this blurb][1]:

> Ordering is only applied during upgrades, where some components rely on another component being updated first. As a convenience, the CVO guarantees that components at an earlier run level will be created or updated before your component is invoked. Note however that components without `ClusterOperator` objects defined may not be fully deployed when your component is executed, so always ensure your prerequisites know that they must correctly obey the `ClusterOperator` protocol to be available. More sophisticated components should observe the prerequisite `ClusterOperator`s directly and use the `versions` field to enforce safety.

What about [the `Kind` hints][2]?  [The extension restrictions][3]?  [The `create-only` annotation][4]?  The `image-references docs][5]?   Probably also worth linking [the profile enhancement][6], because the CVO will ignore manifests that don't declare themselves to be members of the current profile.

[1]: https://github.com/openshift/cluster-version-operator/blame/69099669668f533fa6659a85c26d607505d8d558/docs/dev/operators.md#L32
[2]: https://github.com/openshift/cluster-version-operator/blob/69099669668f533fa6659a85c26d607505d8d558/docs/dev/operators.md#what-do-i-put-in-manifests
[3]: https://github.com/openshift/cluster-version-operator/blame/69099669668f533fa6659a85c26d607505d8d558/docs/dev/operators.md#L98
[4]: https://github.com/openshift/cluster-version-operator/blame/69099669668f533fa6659a85c26d607505d8d558/docs/dev/operators.md#L98
[5]: https://github.com/openshift/cluster-version-operator/blob/69099669668f533fa6659a85c26d607505d8d558/docs/dev/operators.md#how-do-i-ensure-the-right-images-get-used-by-my-manifests
[6]: https://github.com/openshift/enhancements/blob/ffe55d0c2bb63b52b421fb567de66a62e567a04f/enhancements/update/cluster-profiles.md